examples/objectServerExample/src/main/java/io/realm/examples/objectserver/CounterActivity.kt

```diff
@@ -0,0 +1,175 @@
+
+    private val loggedInUser: SyncUser?
+        get() {
+            var user: SyncUser? = null
+
+            try {
+                user = SyncUser.current()
+            } catch (ignore: IllegalStateException) {
+            }
+
+            if (user == null) {
+                startActivity(Intent(this, LoginActivity::class.java))
+            }
+
+            return user
+        }
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_counter)
+        counterView = binding.textCounter
+        progressBar = binding.progressbar
+        binding.upper.setOnClickListener { adjustCounter(1) }
+        binding.lower.setOnClickListener { adjustCounter(-1) }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        user = loggedInUser
+        val user = user
+        if (user != null) {
+            // Create a RealmConfiguration for our user
+            val config = user.createConfiguration(BuildConfig.REALM_URL)
+                    .initialData { realm -> realm.createObject<CRDTCounter>(user.identity) }
+                    .build()
+
+            // This will automatically sync all changes in the background for as long as the Realm is open
+            realm = Realm.getInstance(config)
+
+            counterView.text = "-"
+            counters = realm.where<CRDTCounter>().equalTo("name", user.identity).findAllAsync()
+            counters.addChangeListener { counters, _ ->
+                if (counters.isValid && !counters.isEmpty()) {
+                    val counter = counters.first()
+                    counterView.text = String.format(Locale.US, "%d", counter!!.count)
+                } else {
+                    counterView.text = "-"
+                }
+            }
+
+            // Setup progress listeners for indeterminate progress bars
+            session = realm.syncSession
+            session.run {
+                addDownloadProgressListener(ProgressMode.INDEFINITELY, downloadListener)
+                addUploadProgressListener(ProgressMode.INDEFINITELY, uploadListener)
+            }
+        }
+    }
+
+    override fun onStop() {
+        super.onStop()
+        user?.run {
+            session.run {
+                removeProgressListener(downloadListener)
+                removeProgressListener(uploadListener)
+            }
+            realm.close()
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.menu_counter, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.action_logout -> {
+                realm.close()
+                val user = user
+                if (user != null) {
```

Comment:
The != null isn't so much "the Kotlin way". Have a look at ?.let instead (https://discuss.kotlinlang.org/t/let-vs-if-not-null/3542)

Reply:
Actually it depends on context. `user?.let { user ->` works too, but then the IDE nags you about name shadowing.

Reply:
You don't have to call it user. Renaming it might make it more clear and get around the nag without using the != maybe?

Anyway, it's not a big deal. Just not what I'm used to seeing.

Reply:
I suspect this is one of those small details the Kotlin community is still deciding on what is considered "best practice". I started out using `user?.let { it }` as well, but you can then argue that it can be hard to deduce what `it` is. Giving it a name obviously fixes that, but then the name is different from the variable.

I don't have strong opinions either way and as you point out there might be different trade-offs. I suspect this is just one of those things that each team has to decide on their own until the community gravitates towards a "default" way of doing this.